compiler/aarch64/codegen/OMRCodeGenerator.cpp

```diff
@@ -574,6 +576,54 @@ int64_t OMR::ARM64::CodeGenerator::getSmallestPosConstThatMustBeMaterialized()
    return 0;
    }
 
+
+bool OMR::ARM64::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::ILOpCode opcode, TR::DataType dt)
+   {
+   // implemented vector opcodes
+   switch (opcode.getOpCodeValue())
+      {
+      case TR::vadd:
+      case TR::vsub:
+         if (dt == TR::Int64)
+            return true;
+         /* other types -- fall through */
+      case TR::vmul:
+         if (dt == TR::Int8 || dt == TR::Int16 || dt == TR::Int32)
+            return true;
+         /* other types -- fall through */
+      case TR::vdiv:
+         if (dt == TR::Float || dt == TR::Double)
+            return true;
+         else
+            return false;
+      case TR::vneg:
+         if (dt == TR::Int16 || dt == TR::Float || dt == TR::Double)
```

Comment:
You can add support for vneg Int32/Int64.

Reply:
I did not add that because `OMR::ARM64::TreeEvaluator::vnegEvaluator()` does not have support for Int32/Int64.

Reply:
I meant implementing Int32/Int64 support in `vnegEvaluator()`.  You can do it in a separate PR later.

Reply:
Sure.